sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java

```diff
@@ -102,4 +103,29 @@ public ShareCreateOptions setRootSquash(ShareRootSquash rootSquash) {
         this.rootSquash = rootSquash;
         return this;
     }
+
+    /**
+     * Get the enableSnapshotVirtualDirectoryAccess property: The EnableSnapshotVirtualDirectoryAccess property.
+     * Optional. Supported in version 2023-08-03 and above.
+     * Specifies whether the snapshot virtual directory should be accessible at the root of share mount point when NFS is enabled.
+     * If not specified, the default is true.
+     * @return the enableSnapshotVirtualDirectoryAccess value.
+     */
+    public Boolean isEnableSnapshotVirtualDirectoryAccess() {
```

Comment:
For the handwritten classes feel free to use names that read well even if it doesn't exactly match generated code.

`isSnapshotVirtualDirectoryAccessEnabled`